ome/figure
# Rois from omero

This allows loading of ROIs from OMERO to add to figure panels as new Shapes.

To test:
 - Pick a multi-plane image (also good to test with single plane image)
 - Create a bunch of ROIs with single or multiple shapes of all types using Insight. Include rotated Ellipses.
 - Add that image to OMERO.figure, select it and in the right tab 'ROIs' click "Draw".
 - Test that drawing shapes manually works OK as before.
 - Click "Load ROIs from OMERO"
 - Should see a list of ROIs / shapes, similar to view in webclient with ROIs expandable to show Shapes.
 - Hovering over each Shape row displays it on the image. Clicking on row sets the Z & T of the image to the Z & T of that shape. Clicking "Add" will add the Shape to the image.
 - Click OK to close dialog and save shapes to image panel.
 - Open the "Draw" dialog for a different image to check the dialog is re-rendered correctly.

Discussed with @bramalingam: various things to improve:

 - [x] "Load ROIs" button instead of loading instantly
 - [x] Get bounding box for Ellipse, Line & Rectangle - zoom to show shape within region
 - [x] Collapse shapes under ROI by default (as in webclient & crop dialog)
 - [x] Clicking on shape updates Z/T to that plane (as in webclient) but we don't save Z/T yet.
 - [x] Clicking on "Add Shape" adds shape to viewer and selects it. We save this Z/T.
 - [x] Divergence of shape-editor model and OMERO model. E.g. ellipse cx, cy, rx, ry -> x, y, radiusX, radiusY. Do we update shape-editor to use new model? If so then we need to update existing OMERO.figure files when opened. BUT still need to handle pre OMERO 5.3 with older model.

Follow up ideas:
 - Render all shapes if number of planes is small (10 or less)
 - Maybe option to filter shapes by current plane?
 - When Ellipses are added, x, y, rx, ry are recalculated from handles (to handle transforms) but this means that they are not detected as duplicates
 - Maybe add Z & T sliders?
 - Maybe show shapes on viewer when mouseover area!? - click to add?!
 - Do we allow shape to be added if previously added & moved? Don't want duplicate IDs!
 - Zoom out of preview to show shape in context? Make line thicker to improve visibility?
